In [98]:
import pandas as pd
import numpy as np
import requests

# contest.list

Returns information about all available contests.

### Parameter	Description
gym	Boolean. If true — than gym contests are returned. Otherwide, regular contests are returned.
Return value: Returns a list of Contest objects. If this method is called not anonymously, then all available contests for a calling user will be returned too, including mashups and private gyms.

In [99]:
data=requests.get("https://codeforces.com/api/contest.list?").json()
data1=pd.DataFrame(data['result'])
data1

id                                               name  type     phase   
0     1827                          Codeforces Round (Div. 1)    CF    BEFORE  \
1     1828                          Codeforces Round (Div. 2)    CF    BEFORE   
2     1832  Educational Codeforces Round 148 (Rated for Di...  ICPC    BEFORE   
3     1824                          Codeforces Round (Div. 1)    CF    BEFORE   
4     1825                          Codeforces Round (Div. 2)    CF    BEFORE   
...    ...                                                ...   ...       ...   
1734     5                            Codeforces Beta Round 5  ICPC  FINISHED   
1735     4              Codeforces Beta Round 4 (Div. 2 Only)  ICPC  FINISHED   
1736     3                            Codeforces Beta Round 3  ICPC  FINISHED   
1737     2                            Codeforces Beta Round 2  ICPC  FINISHED   
1738     1                            Codeforces Beta Round 1  ICPC  FINISHED   

      frozen  durationSeconds  startTimeSeconds  relativeTimeSeconds  
0      False             7200        1684074900              -945230  
1      False             7200        1684074900              -945230  
2      False             7200        1683902100              -772430  
3      False             7200        1683547500              -417830  
4      False             7200        1683547500              -417830  
...      ...              ...               ...                  ...  
1734   False             7200        1269100800            414028870  
1735   False             7200        1268395200            414734470  
1736   False             7200        1267963200            415166470  
1737   False             7200        1267117200            416012470  
1738   False             7200        1266580800            416548870  

[1739 rows x 8 columns]

# contest.ratingChanges
Returns rating changes after the contest.

## Parameter	Description
contestId (Required)	Id of the contest. It is not the round number. It can be seen in contest URL.   
For example: /contest/566/status
Return value: Returns a list of RatingChange objects.  

Example: https://codeforces.com/api/contest.ratingChanges?contestId=566

Note: 760*1737 *1141= 1.5 GB

In [100]:
data=requests.get("https://codeforces.com/api/contest.ratingChanges?contestId=5").json()
data1=pd.DataFrame(data['result'])
data1

contestId              contestName       handle  rank   
0            5  Codeforces Beta Round 5         Petr     1  \
1            5  Codeforces Beta Round 5      ilyaraz     2   
2            5  Codeforces Beta Round 5      RAVEman     3   
3            5  Codeforces Beta Round 5       vlad89     4   
4            5  Codeforces Beta Round 5      ilyakor     5   
..         ...                      ...          ...   ...   
327          5  Codeforces Beta Round 5  khaled.hady   315   
328          5  Codeforces Beta Round 5        AnhDQ   315   
329          5  Codeforces Beta Round 5          GDR   315   
330          5  Codeforces Beta Round 5    Batyr_004   315   
331          5  Codeforces Beta Round 5          ika   315   

     ratingUpdateTimeSeconds  oldRating  newRating  
0                 1269108000       1775       1891  
1                 1269108000       1500       1720  
2                 1269108000       1655       1813  
3                 1269108000       1644       1803  
4                 1269108000       1762       1876  
..                       ...        ...        ...  
327               1269108000       1500       1391  
328               1269108000       1500       1391  
329               1269108000       1357       1276  
330               1269108000       1384       1298  
331               1269108000       1610       1476  

[332 rows x 7 columns]

## contest.standings

Returns the description of the contest and the requested part of the standings.

Parameter	Description  
contestId (Required)	Id of the contest. It is not the round number. It can be seen in contest URL. For example: /contest/566/status  
from	1-based index of the standings row to start the ranklist.  
count	Number of standing rows to return.  
handles	Semicolon-separated list of handles. No more than 10000 handles is accepted.  
room	If specified, than only participants from this room will be shown in the result. If not — all the participants will be shown.  
showUnofficial	If true than all participants (virtual, out of competition) are shown. Otherwise, only official contestants are shown.  
Return value: Returns object with three fields: "contest", "problems" and "rows".   
Field "contest" contains a Contest object.   
Field "problems" contains a list of Problem objects. Field "rows" contains a list of RanklistRow objects.

In [103]:
data=requests.get("https://codeforces.com/api/contest.standings?contestId=566&showUnofficial=true").json()
data1=pd.DataFrame(data['result']['problems'])
data1

contestId index                              name         type  points   
0        566     A                    Matching Names  PROGRAMMING  1750.0  \
1        566     B             Replicating Processes  PROGRAMMING  2500.0   
2        566     C              Logistical Questions  PROGRAMMING  3000.0   
3        566     D             Restructuring Company  PROGRAMMING  1000.0   
4        566     E                     Restoring Map  PROGRAMMING  3000.0   
5        566     F  Clique in the Divisibility Graph  PROGRAMMING   500.0   
6        566     G                       Max and Min  PROGRAMMING  2500.0   

   rating                                          tags  
0    2300             [dfs and similar, strings, trees]  
1    2600             [constructive algorithms, greedy]  
2    3000  [dfs and similar, divide and conquer, trees]  
3    1900                        [data structures, dsu]  
4    3200    [bitmasks, constructive algorithms, trees]  
5    1500                     [dp, math, number theory]  
6    2500                                    [geometry]

In [104]:
data=requests.get("https://codeforces.com/api/contest.standings?contestId=566&showUnofficial=true").json()
data1=pd.DataFrame(data['result']['rows'])
data1

party  rank  points   
0     {'contestId': 566, 'members': [{'handle': 'mar...     1  8238.0  \
1     {'contestId': 566, 'members': [{'handle': 'rng...     2  7974.0   
2     {'contestId': 566, 'members': [{'handle': 'hog...     3  7048.0   
3     {'contestId': 566, 'members': [{'handle': 'ngf...     4  6011.0   
4     {'contestId': 566, 'members': [{'handle': 'ste...     5  5872.0   
...                                                 ...   ...     ...   
6055  {'contestId': 566, 'members': [{'handle': 'yin...     0     0.0   
6056  {'contestId': 566, 'members': [{'handle': 'WII...     0     0.0   
6057  {'contestId': 566, 'members': [{'handle': 'dmi...     0     0.0   
6058  {'contestId': 566, 'members': [{'handle': 'ser...     0     0.0   
6059  {'contestId': 566, 'members': [{'handle': 'MHD...     0     0.0   

      penalty  successfulHackCount  unsuccessfulHackCount   
0           0                    0                      0  \
1           0                    1                      0   
2           0                    0                      0   
3           0                    3                      1   
4           0                    0                      0   
...       ...                  ...                    ...   
6055        0                    0                      0   
6056        0                    0                      0   
6057        0                    0                      0   
6058        0                    0                      0   
6059        0                    0                      0   

                                         problemResults  
0     [{'points': 1582.0, 'rejectedAttemptCount': 0,...  
1     [{'points': 1330.0, 'rejectedAttemptCount': 0,...  
2     [{'points': 1162.0, 'rejectedAttemptCount': 0,...  
3     [{'points': 1267.0, 'rejectedAttemptCount': 0,...  
4     [{'points': 1568.0, 'rejectedAttemptCount': 0,...  
...                                                 ...  
6055  [{'points': 0.0, 'rejectedAttemptCount': 0, 't...  
6056  [{'points': 0.0, 'rejectedAttemptCount': 0, 't...  
6057  [{'points': 0.0, 'rejectedAttemptCount': 0, 't...  
6058  [{'points': 0.0, 'rejectedAttemptCount': 0, 't...  
6059  [{'points': 0.0, 'rejectedAttemptCount': 0, 't...  

[6060 rows x 7 columns]

In [105]:
data=requests.get("https://codeforces.com/api/contest.standings?contestId=566&showUnofficial=true").json()
data1=pd.DataFrame(data['result']['rows'][0]['party'])
data1

contestId                 members participantType  ghost  startTimeSeconds
0        566  {'handle': 'maroonrk'}         VIRTUAL  False        1621605600

In [106]:
data=requests.get("https://codeforces.com/api/contest.standings?contestId=566&showUnofficial=true").json()
data1=pd.DataFrame(data['result']['rows'][0]['problemResults'])
data1

points  rejectedAttemptCount   type  bestSubmissionTimeSeconds
0  1582.0                     0  FINAL                     1476.0
1  1830.0                     0  FINAL                     4078.0
2  1678.0                     1  FINAL                     6412.0
3   872.0                     0  FINAL                     1953.0
4     0.0                     5  FINAL                        NaN
5   446.0                     0  FINAL                     1665.0
6  1830.0                     2  FINAL                     3445.0